### PR TITLE
CVP-1686: Fixed SonarQube blocker and major issues

### PR DESCRIPTION
**Pull request by 007sunny**
SonarQube issues: https://sonarqube.services.icanbwell.com/project/issues?id=icanbwell_fhir-server&issues=AYWX0PUcn0zoDOI5r3_E&open=AYWX0PUcn0zoDOI5r3_E

### CODE REVIEW

File: src/utils/bwellPersonFinder.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ class BwellPersonFinder {
         let linkedPersons = await databaseQueryManager.findAsync({ query: { 'link.target.reference': currentSubject }});
 
         // iterate over linked Persons (breadth search)
-        while (!foundPersonId && linkedPersons && await linkedPersons.hasNext()) {
+        while (!foundPersonId && await linkedPersons.hasNext()) {
```
Comment (anmolmittal1) [inline src/utils/bwellPersonFinder.js:60]: We should add a if (linkedPersons) check around this while loop

File: src/graphql/v1/generator/fhir_xml_schema_parser.py [hunk 2]
```diff
@@ -757,12 +757,10 @@ def generate_properties_for_class(
...
-                        cleaned_type=cleaned_type
-                        if cleaned_type not in FhirXmlSchemaParser.cleaned_type_mapping
-                        else FhirXmlSchemaParser.cleaned_type_mapping[cleaned_type],
-                        type_snake_case=FhirXmlSchemaParser.camel_to_snake(cleaned_type)
-                        if cleaned_type not in FhirXmlSchemaParser.cleaned_type_mapping
-                        else FhirXmlSchemaParser.camel_to_snake(cleaned_type),
+                        cleaned_type=FhirXmlSchemaParser.cleaned_type_mapping.get(cleaned_type, cleaned_type),
+                        type_snake_case=FhirXmlSchemaParser.camel_to_snake(
+                            FhirXmlSchemaParser.cleaned_type_mapping.get(cleaned_type, cleaned_type)
+                        ),
```
Comment (anmolmittal1) [inline src/graphql/v1/generator/fhir_xml_schema_parser.py:763]: this should directly be
`type_snake_case=FhirXmlSchemaParser.camel_to_snake(cleaned_type)`

**Review by anmolmittal1 — CHANGES_REQUESTED** [review]

File: src/utils/bwellPersonFinder.js (hunk 1 above)
Comment (007sunny) [reply src/utils/bwellPersonFinder.js:60]: linkedPersons won't be null. It's cursor even if the results are empty.

File: src/graphql/v1/generator/fhir_xml_schema_parser.py (hunk 2 above)
Comment (anmolmittal1) [reply src/graphql/v1/generator/fhir_xml_schema_parser.py:763]: No, The error at sonarqube is saying that the value is same no matter what. Here we seem to have changed the logic that we are passing `FhirXmlSchemaParser.cleaned_type_mapping.get(cleaned_type, cleaned_type)` as argument to `FhirXmlSchemaParser.camel_to_snake`. Previously it we were only passing `cleaned_type` as arg.

**Pull request merged**